### ISSUE TITLE
(Help!) I can't catch pokemon (There's a red error!)

### ISSUE THREAD

**Issue by harambeman52** [finndev/PokeBuddy#675]
I have been going strong ever since yesterday. But now these "glitches" have happened:
-Shows me with , lvl 0 with 0/0 experience (even though i am lvl 29)
-Red error tasktask text shows every time I try to catch a lured and or regular pokemon
HOWEVER, pokestops work fine and I receive the regular 50 experience per.
Just curious on what I can do to catch pokemon again, remove the red text and make it properly say my dust, experience, level, etc.
WHAT I HAVE TRIED ALREADY: Uninstalling/reinstalling, trying a new account, trying google instead of PTC.
As well as I dont have 250/250 pokemon and I have enough pokeballs.

Unfortunately none of these have worked so I decided to post here!
tl;dr getting red error text when bot tries to catch pokemon, pokestops work fine
also doesnt recylce items, transfer pokemon, etc.

Looks like this: http://prntscr.com/c3r8m1
(red text didnt spawn yet)

Red text example:
P - 19:30:26] [Farm Pokestops - ERROR]: System.ArgumentNullException: Value cannot be null.
Parameter name: source
   at System.Linq.Enumerable.FirstOrDefault[TSource](IEnumerable`1 source, Func`2 predicate)
   at dwb7:\.\*$/#)|\.XWp_|VK}\.^n$.RnfK|\]re^\.!qA1w%r;cFze'\]$.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at dwb7:\.\*$/#)|\.XWp_|VK}\.^n$.f/KXD`P^<M`(L\\%-Pvf{EI>K'.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task task)
   at PokeBuddy.Niantic.Behaviours.CatchPokemons.<CatchPokemon>d__15.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at PokeBuddy.Niantic.Behaviours.CatchPokemons.<CatchLurePokemon>d__14.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task task)
   at PokeBuddy.Niantic.Behaviours.FarmPokeStops.<Execute>d__13.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at PokeBuddy.Core.Routine.<>c.<<Start>b__16_3>d.MoveNext()

**Comment by cocainecowboy24** [comment]
Hi Devs,

First off amazing project. I'm a huge fan.

Ever since the new update, I am also experiencing the same problem. PokeBuddy allows me to sign in but does not show my level at all (shows 0) and does not show that I have any items or pokeballs on the main screen although while farming pokestops my bag still reports full. Additionally, I cannot catch pokemon at all. It says found but never actually tries to catch any. I am only use one account per this ip address.


Your help is much appreciated. Thanks Again

**Comment by finndev** [comment]
It seems some "special" case accounts cause issues, because I can't reproduce it, and only a very low amount of users have it. If you want me to fix it, email me your account (email visible in logwindow) where this bug occurs and I will fix it.

**Comment by curtiola** [comment]
Sometimes it doesnt update correctly, try downloading the new release? Or is it possible you just need to restart your pc? It may sound remedial but it works for some issues.

**Comment by harambeman52** [comment]
Could you rephrase what you mean by logwindow? Like the red text? + account credentials?

**Comment by finndev** [comment]
@harambeman52 In the log window, when you start the bot, you see an email. Send your account credentials to it if you want me to fix it. I'll only use the account for a few minutes (about 30), send me your current GPS Coords as well, so you won't get soft-banned.

**Comment by curtiola** [comment]
The first entry in the console on the bottom will show you the email address

**Comment by harambeman52** [comment]
Ok -- will do! Thanks!

**Comment by finndev** [comment]
@harambeman52 Thanks, will you do it now? Just asking, because I won't be online for long, but if you do it now, I'll fix it now.

**Comment by harambeman52** [comment]
Yeah, I am composing the email right now should be sent within 2-3 minutes.

**Comment by harambeman52** [comment]
I dont care about the soft ban either so I didn't send coords.

**Comment by harambeman52** [comment]
Although if you absolutely need them I can relog on and get them.

**Comment by finndev** [comment]
@harambeman52 Checking now, thanks for the account

**Comment by TostiJostii** [comment]
finndev I have almost the same problem. I just stopped a 13 hour farming and now it shows that i'm lvl 0 and won't catch any pokemon. Do you have an idea what is wrong?

**Comment by finndev** [comment]
@harambeman52 OK, it doesn't seem to be account related. I can bot just fine on your account.

![](http://i.imgur.com/b9X3im7.png)

Are you using last version?

**Comment by harambeman52** [comment]
Yeah I was using the latest version. Could it have anything to do with my internet? Other things I could think of are: Corrupted files (idk), virus scanners somehow breaking it, etc.
Other then that I don't know.

**Comment by harambeman52** [comment]
I might trying completely purging pokebuddy from my computer and try reinstalling it again. Will post back if it fixes it that way.

**Comment by curtiola** [comment]
Latest version via update or download? Could just need to redownload, sometimes it doesn't update properly, or try restart pc?
I posted the same above, wasn't sure you saw/tried it

**Comment by cocainecowboy24** [comment]
Hey FinnDev,

After trying everything in the world to fix this in 3 hours, all I had to do was restart my computer like an idiot and it worked!!!!!! I CANT BELIEVE IT!!!!!! RESTART YOUR COMPUTER IF YOU ARE HAVING THIS ISSUE!!!!!!!

Finn Dev you are the best. Thank you for amazing code and quick response.

Curtiola thank you for the idea lol

**Comment by curtiola** [comment]
No problem xD glad I could help

**Comment by harambeman52** [comment]
Should I restart my computer after or before I download it again?

**Comment by curtiola** [comment]
If you already purged it then download first

**Comment by cocainecowboy24** [comment]
I restarted after the download and worked perfectly immediately after LOLOLOLOLOLOLOLOLOLOLOLOL

**Comment by curtiola** [comment]
Otherwise restart first if you haven't deleted it

**Comment by harambeman52** [comment]
K I will restart after I download and see if it works!

**Comment by TostiJostii** [comment]
Ow my fucking god THANK YOU

**Comment by harambeman52** [comment]
Hmm. It worked! Not sure if it had something to do with an old "Dependencies" file or not but restarting my computer did the job. Thanks everyone! And thanks for the help Finndev for taking the time out your day to help me!

**Comment by finndev** [comment]
What a weird bug. Unexplainable, but ok

**Comment by curtiola** [comment]
Woo-hoo haha I'm glad it worked
